quickstarts/microsoft.machinelearningservices/machine-learning-workspace/main.bicep

```diff
@@ -0,0 +1,114 @@
+@description('Specifies the name of the deployment.')
+param name string
+
+@description('Specifies the name of the environment.')
+param environment string
+
+@description('Specifies the location of the Azure Machine Learning workspace and dependent resources.')
+@allowed([
+  'australiaeast'
+  'brazilsouth'
+  'canadacentral'
+  'centralus'
+  'eastasia'
+  'eastus'
+  'eastus2'
+  'francecentral'
+  'japaneast'
+  'koreacentral'
+  'northcentralus'
+  'northeurope'
+  'southeastasia'
+  'southcentralus'
+  'uksouth'
+  'westcentralus'
+  'westus'
+  'westus2'
+  'westeurope'
+  'usgovvirginia'
+])
+param location string
+
+var tenantId = subscription().tenantId
+var storageAccountName = 'st${name}${environment}'
+var keyVaultName = 'kv-${name}-${environment}'
+var applicationInsightsName = 'appi-${name}-${environment}'
+var containerRegistryName = 'cr${name}${environment}'
+var workspaceName = 'mlw${name}${environment}'
+var storageAccountId = storageAccount.id
+var keyVaultId = vault.id
+var applicationInsightId = applicationInsight.id
+var containerRegistryId = registry.id
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2022-05-01' = {
+  name: storageAccountName
+  location: location
+  sku: {
+    name: 'Standard_RAGRS'
+  }
+  kind: 'StorageV2'
+  properties: {
+    encryption: {
+      services: {
+        blob: {
+          enabled: true
+        }
+        file: {
+          enabled: true
+        }
+      }
+      keySource: 'Microsoft.Storage'
+    }
+    supportsHttpsTrafficOnly: true
+    allowBlobPublicAccess: false
+  }
+}
+
+resource vault 'Microsoft.KeyVault/vaults@2022-07-01' = {
+  name: keyVaultName
+  location: location
+  properties: {
+    tenantId: tenantId
+    sku: {
+      name: 'standard'
+      family: 'A'
+    }
+    accessPolicies: []
+    enableSoftDelete: true
+  }
+}
+
+resource applicationInsight 'Microsoft.Insights/components@2020-02-02' = {
+  name: applicationInsightsName
+  location: (((location == 'eastus2') || (location == 'westcentralus')) ? 'southcentralus' : location)
```

Comment:
I know this isn't really the point of your PR, but I think we need to retire this pattern, here's what we should do instead: https://github.com/Azure/azure-quickstart-templates/commit/f8c766ab21d9b12e62c8778e6a696e0b57aa1aa6

For the most part AI is available everywhere, but sets are actually intersecting and the intersecting part is shrinking.  So eventually not giving user control over this will cause these to fail and they won't be able to fix it without modifying.  I'd like to get it while we're going after these...